src/main/java/org/kuali/kfs/coa/document/validation/impl/AccountGlobalRule.java

```diff
@@ -863,5 +962,1065 @@ private boolean isIncomeStreamAccountRequired(String fundGroupCode, String subFu
+        			for(AccountGlobalDetail accountGlobalDetail : newAccount.getAccountGlobalDetails()){
+        				accountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+        				success &= !accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts().isEmpty();
+        				if(success){
+        					continue;
+        				}
+        			}
+        		}
+        		
+        	}
+        }
+        
+        if(!expectFilled){
+        	 success = newAccount.getActiveIndirectCostRecoveryAccounts().isEmpty();
+        	 
+        	 if (!success){
+                 putFieldError(KFSPropertyConstants.INDIRECT_COST_RECOVERY_ACCOUNTS, errorMessage, args);
+             }
+        	  if(success){
+        		  if(ObjectUtils.isNotNull(newAccount.getAccountGlobalDetails()) && !newAccount.getAccountGlobalDetails().isEmpty()){
+          			for(AccountGlobalDetail accountGlobalDetail : newAccount.getAccountGlobalDetails()){
+          				accountGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+          				boolean empty = accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts().isEmpty();
+          				if(!empty){
+          					 putFieldError(KFSPropertyConstants.ACCOUNT_CHANGE_DETAILS, CUKFSKeyConstants.ERROR_DOCUMENT_ACCT_GLB_MAINT_ACCOUNT_CG_FIELDS_FILLED_FOR_NON_CG_ACCOUNT, new String[] { args, accountGlobalDetail.getAccountNumber() });
+          				   
+          				}
+          				success &= accountGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts().isEmpty();
+          				if(success){
+          					continue;
+          				}
+          			}
+        		  }
+        	  }
+        			
+        }
+       
+        if (!success){
+            putFieldError(KFSPropertyConstants.INDIRECT_COST_RECOVERY_ACCOUNTS, errorMessage, args);
+        }
+        return success;
+    }
+    
+    /**
+     * This method checks if the ICR collection should or should not be filled
+     * add error message if validation is not successful
+     * 
+     * @param expectFilled
+     * @param errorMessage
+     * @param args
+     * @return
+     */
+    protected boolean checkICRCollectionExistWithErrorMessage(boolean expectFilled, String errorMessage, String[] args, CuAccountGlobal newAccount, AccountGlobalDetail acctGlobalDetail) {
+		boolean success = true;
+		if (expectFilled) {
+			success = !newAccount.getActiveIndirectCostRecoveryAccounts().isEmpty();
+			if (success) {
+				for (IndirectCostRecoveryAccountChange account : newAccount.getActiveIndirectCostRecoveryAccounts()) {
+					success &= StringUtils.isNotBlank(account.getIndirectCostRecoveryAccountNumber()) && StringUtils.isNotBlank(account.getIndirectCostRecoveryFinCoaCode());
+				}
+			}
+
+			if (!success) {
+				acctGlobalDetail.refreshReferenceObject(KFSPropertyConstants.ACCOUNT);
+
+				success &= !acctGlobalDetail.getAccount().getActiveIndirectCostRecoveryAccounts().isEmpty();
+			}
```

Comment:
success might be true as it checks for active icr accounts on the account in the account global detail (on the account being editted). Initially it checked if any icr accounts were entered on the Account global doc.